src/axom/primal/tests/primal_solid_angle.cpp

```diff
@@ -478,6 +478,195 @@ TEST(primal_solid_angle, selfintersecting_quadrilateral)
+    Point3D {        0, 4*(1-2*rt3)/3, 4*(1-2*rt3)/3}, Point3D {          0, rt2*(2*rt3-7)/3,    -5*rt6/3}, Point3D {0,               0,   4*(rt3-5)/3}, Point3D {          0, rt2*(7-2*rt3)/3,    -5*rt6/3}, Point3D {        0, 4*(2*rt3-1)/3, 4*(1-2*rt3)/3},
+    Point3D {      rt2, rt2*(rt3 - 4), rt2*(rt3 - 4)}, Point3D {(3*rt3-2)/2,     (2-3*rt3)/2,  -(rt3+6)/2}, Point3D {rt2*(7-2*rt3)/3, 0,      -5*rt6/3}, Point3D {(3*rt3-2)/2,   (3*rt3-2)/2,    -(rt3+6)/2}, Point3D {      rt2,   rt2*(4-rt3),   rt2*(rt3-4)},
+    Point3D {4*(rt3-1),     4*(1-rt3),     4*(1-rt3)}, Point3D {rt2*(4-rt3),            -rt2, rt2*(rt3-4)}, Point3D {4*(2*rt3-1)/3,   0, 4*(1-2*rt3)/3}, Point3D {rt2*(4-rt3),           rt2,   rt2*(rt3-4)}, Point3D {4*(rt3-1),     4*(rt3-1),     4*(1-rt3)}};
+
+  double weight_data[25] = {
+         4*(3-rt3), rt2*(3*rt3-2),   4*(5-rt3)/3, rt2*(3*rt3-2),     4*(3-rt3),
+     rt2*(3*rt3-2),     (rt3+6)/2, rt2*(rt3+6)/3,     (rt3+6)/2, rt2*(3*rt3-2),
+       4*(5-rt3)/3, rt2*(rt3+6)/3, 4*(5*rt3-1)/9, rt2*(rt3+6)/3,   4*(5-rt3)/3,
+     rt2*(3*rt3-2),     (rt3+6)/2, rt2*(rt3+6)/3,     (rt3+6)/2, rt2*(3*rt3-2),
+         4*(3-rt3), rt2*(3*rt3-2),   4*(5-rt3)/3, rt2*(3*rt3-2),     4*(3-rt3)
+  };
+  // clang-format on
+
+  BPatch sphere_faces[6];
+  for(int n = 0; n < 6; ++n)
+  {
+    sphere_faces[n].setOrder(4, 4);
+    sphere_faces[n].makeRational();
+  }
+
+  sphere_faces[0].setOrder(4, 4);
+  for(int i = 0; i < 5; ++i)
+  {
+    for(int j = 0; j < 5; ++j)
+    {
+      int idx = 5 * i + j;
+      for(int n = 0; n < 6; ++n)
+        sphere_faces[n].setWeight(i, j, weight_data[idx]);
+
+      // Set up each face
+      sphere_faces[0](i, j)[0] = node_data[idx][1] / weight_data[idx];
+      sphere_faces[0](i, j)[1] = node_data[idx][0] / weight_data[idx];
+      sphere_faces[0](i, j)[2] = node_data[idx][2] / weight_data[idx];
+
+      sphere_faces[1](i, j)[0] = -node_data[idx][0] / weight_data[idx];
+      sphere_faces[1](i, j)[1] = -node_data[idx][1] / weight_data[idx];
+      sphere_faces[1](i, j)[2] = -node_data[idx][2] / weight_data[idx];
+
+      sphere_faces[2](i, j)[0] = node_data[idx][2] / weight_data[idx];
+      sphere_faces[2](i, j)[1] = node_data[idx][1] / weight_data[idx];
+      sphere_faces[2](i, j)[2] = node_data[idx][0] / weight_data[idx];
+
+      sphere_faces[3](i, j)[0] = -node_data[idx][1] / weight_data[idx];
+      sphere_faces[3](i, j)[1] = -node_data[idx][2] / weight_data[idx];
+      sphere_faces[3](i, j)[2] = -node_data[idx][0] / weight_data[idx];
+
+      sphere_faces[4](i, j)[0] = node_data[idx][0] / weight_data[idx];
+      sphere_faces[4](i, j)[1] = node_data[idx][2] / weight_data[idx];
+      sphere_faces[4](i, j)[2] = node_data[idx][1] / weight_data[idx];
+
+      sphere_faces[5](i, j)[0] = -node_data[idx][2] / weight_data[idx];
+      sphere_faces[5](i, j)[1] = -node_data[idx][0] / weight_data[idx];
+      sphere_faces[5](i, j)[2] = -node_data[idx][1] / weight_data[idx];
+    }
+  }
+
```

Comment:
Did we set up integrals over a `BezierPatch`? If so, it'd be neat to check that the integral is $4/3\cdot\pi r^3$

Reply:
I had written a function (with tests) to do this integral, but when the winding number algorithm shifted to using Stokes' theorem, it ended up being unused. Since it isn't related to or needed by our 3D winging number algorithm anymore, I've decided to push it to another branch down the line that will do a little more with our spherical patch surface.